plugins/tests/test_mapcss_parsing_evaluation.validator.mapcss

```diff
@@ -561,3 +561,34 @@ node[to_float("3.14e1") = 31.4][to_double("3.1415e1") = 31.415] {
   throwWarning: "test";
   assertMatch: "node x=y";
 }
+
+node[tag_length][siunit_length(tag(tag_length)) == "1.2"] {
+  throwWarning: tr("test unit conversion {0} -> {1}", "{0.value}", siunit_length(tag(tag_length)));
+  assertMatch: "node tag_length=1.2";
+}
+node[tag_length][siunit_length(tag(tag_length)) == tag(b)] {
+  throwWarning: tr("test unit conversion {0} -> {1}", "{0.value}", siunit_length(tag(tag_length)));
+  assertMatch: "node tag_length=\"  1,3 m \" b=1.3";
+  assertMatch: "node tag_length=1.4m         b=1.4";
+  assertMatch: "node tag_length=1.4meter     b=1.4";
+  assertMatch: "node tag_length=150cm        b=1.5";
+  assertMatch: "node tag_length=1600.0mm     b=1.6";
+  assertMatch: "node tag_length=1.7km        b=1700.0";
+  assertMatch: "node tag_length=1.7kilometer b=1700.0";
+  assertMatch: "node tag_length=-1.8km       b=-1800.0";
+  assertMatch: "node tag_length=10ft         b=3.048";
+  assertMatch: "node tag_length=20'          b=6.096";
+  assertMatch: "node tag_length=100in        b=2.54";
+  assertMatch: "node tag_length=200\"        b=5.08";
+  assertMatch: "node tag_length=1nmi         b=1852.0";
+  assertMatch: "node tag_length=1mi          b=1609.344";
+  assertMatch: "node tag_length=\"10ft 1in\" b=3.0734";
+  assertMatch: "node tag_length=20'2\"       b=6.1468";
+  assertMatch: "node tag_length=-20'2\"      b=-6.1468";
```

Comment:
Do you know if '1,600.0m' is supposed to be supported too?
Or '1 600 m'.

Reply:
The first one is not supposed to be supported: JOSM [replaces](https://josm.openstreetmap.de/browser/josm/trunk/src/org/openstreetmap/josm/tools/Utils.java#L2034) all `,` by `.` so that would become an invalid number (`1.600.0m`). In case they update their code to only do this replacement if there's no `.` in the string yet, we should follow of course, for the mapcss part.

Regarding `1 600 m`: good point, I've updated the code of the mapcss function to match JOSMs behavior here too.

(I didn't do this in the functions in /units.py as I also want to be able to use those for e.g. validation purposes)

Reply:
I see for '1,600.0m', and it makes sense. This looks a bit dangerous for '1,600m', which might mean '1600 m', but it doesn't matter for this mapcss function - it is more a matter of OSM data that should follow guidelines.

Reply:
Yep, well, also with mapcss it can result in false positives if used in selectors (e.g. adding a warning about an object that's too small when the comma was a true 1000-separator) but in that case it also needs to be fixed by JOSM. And in case of such a false positive, at least it'll hopefully make the user think another time about the value...

Of course we can choose to remove this replacal ourselves if desired.